DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -487,6 +498,146 @@ internal DiscordGuild()
 
         #region Guild Methods
 
+        /// <summary>
+        /// Creates a new scheduled event in this guild.
+        /// </summary>
+        /// <param name="name">The name of the event to create, up to 100 characters.</param>
+        /// <param name="description">The description of the event, up to 1000 characters.</param>
+        /// <param name="channelId">If a <see cref="ScheduledGuildEventType.StageInstance"/> or <see cref="ScheduledGuildEventType.VoiceChannel"/>, the id of the channel the event will be hosted in</param>
+        /// <param name="type">The type of the event. <see paramref="channelId"/> must be supplied if not an external event.</param>
+        /// <param name="privacyLevel">The privacy level of thi</param>
+        /// <param name="start">When this event starts. Must be in the future, and before the end date.</param>
+        /// <param name="end">When this event ends. If supplied, must be in the future and after the end date. This is requred for <see cref="ScheduledGuildEventType.External"/>.</param>
+        /// <param name="location">Where this event takes place, up to 100 characters. Only applicable if the type is <see cref="ScheduledGuildEventType.External"/></param>
+        /// <returns>The created event.</returns>
+        public Task<DiscordScheduledGuildEvent> CreateEventAsync(string name, string description, ulong? channelId, ScheduledGuildEventType type, ScheduledGuildEventPrivacyLevel privacyLevel, DateTimeOffset start, DateTimeOffset? end, string location = null)
+        {
+            if (start <= DateTimeOffset.Now)
+                throw new ArgumentOutOfRangeException("The start time for an event must be in the future.");
+
+            if (end != null && end <= start)
+                throw new ArgumentOutOfRangeException("The end time for an event must be after the start time.");
+
+            DiscordScheduledGuildEventMetadata metadata = null;
+
```

Comment:
```suggestion
            switch (type)
            {
                case ScheduledGuildEventType.Stage or ScheduledGuildEventType.Voice when channelId == null:
                    throw new ArgumentException($"{nameof(channelId)} must not be null when type is {type}", nameof(channelId));
                case ScheduledGuildEventType.External when channelId != null:
                    throw new ArgumentException($"{nameof(channelId)} must be null when using external event type", nameof(channelId));
                case ScheduledGuildEventType.External when location == null:
                    throw new ArgumentException($"{nameof(location)} must not be null when using external event type", nameof(location));
                case ScheduledGuildEventType.External when end == null:
                    throw new ArgumentException($"{nameof(endTime)} must not be null when using external event type", nameof(endTime));
            }
```

Reply:
Probably should've verified this code works prior to committing. I'll have to fix it in the morning however